docker/Dockerfile

```diff
@@ -0,0 +1,78 @@
+FROM debian:jessie
+
+ENV PYTHONIOENCODING UTF-8
+
+# Pypy is installed from a package manager because it takes so long to build.
+RUN apt-get update && apt-get install -y \
+    build-essential \
+    curl \
+    git \
+    libbz2-dev \
+    libcurl4-openssl-dev \
+    libmemcached-dev \
+    libncurses5-dev \
+    libreadline-dev \
+    libsqlite3-dev \
+    libssl-dev \
+    pkg-config \
+    pypy \
+    wget \
+    zlib1g-dev
+
+# Install couchbase
+RUN wget http://packages.couchbase.com/clients/c/libcouchbase-2.8.4_jessie_amd64.tar \
+    && tar -vxf libcouchbase-2.8.4_jessie_amd64.tar \
+    && dpkg -i libcouchbase-2.8.4_jessie_amd64/libcouchbase2-core_2.8.4-1_amd64.deb \
+    && dpkg -i libcouchbase-2.8.4_jessie_amd64/libcouchbase-dev_2.8.4-1_amd64.deb
+
+# Setup our home environment
+ENV CELERY_USER someone
+RUN addgroup --gid 1000 ${CELERY_USER}
+RUN adduser --system --disabled-password --uid 1000 --gid 1000 ${CELERY_USER}
+RUN usermod -aG sudo ${CELERY_USER}
+ENV HOME /home/${CELERY_USER}
+WORKDIR ${HOME}
+USER ${CELERY_USER}
+
+# For managing all the local python installations for testing, use pyenv
+RUN curl -L https://raw.githubusercontent.com/pyenv/pyenv-installer/master/bin/pyenv-installer | bash
+
+# Pyenv setup
+ENV PATH="$HOME/.pyenv/bin:$PATH"
+
+# To enable testing versions like 3.4.8 as 3.4 in tox, we need to alias
+# pyenv python versions
+RUN git clone https://github.com/s1341/pyenv-alias.git $(pyenv root)/plugins/pyenv-alias
+
+# Python versions to test against
+RUN VERSION_ALIAS="python2.7" pyenv install 2.7.14
+RUN VERSION_ALIAS="python3.4" pyenv install 3.4.8
+RUN VERSION_ALIAS="python3.5" pyenv install 3.5.5
+RUN VERSION_ALIAS="python3.6" pyenv install 3.6.4
+
+# Install celery
+
+COPY --chown=1000:1000 requirements ${HOME}/requirements
+COPY --chown=1000:1000 docker/entrypoint /entrypoint
+RUN chmod gu+x /entrypoint
+
+# Setup one celery environment for basic development use
+RUN pyenv local python3.6
+RUN pyenv exec pip install \
+  -r requirements/default.txt \
+  -r requirements/pkgutils.txt \
+  -r requirements/test.txt \
+  -r requirements/test-ci-base.txt \
+  -r requirements/test-integration.txt
+
+COPY --chown=1000:1000 MANIFEST.in Makefile setup.py setup.cfg tox.ini ${HOME}/
+COPY --chown=1000:1000 t ${HOME}/t
+COPY --chown=1000:1000 celery ${HOME}/celery
+
+RUN pyenv exec pip install -e .
```

Comment:
The Dockerfile is quite extensive. Perhaps if it was divided in smaller scripts, depending on functionality, it would be better. 

Something like:

```
docker/scripts/create-linux-user.sh
docker/scripts/install-couchbase.sh
docker/scripts/install-pyenv.sh
...
```

What do you think?

Reply:
I agree. I prefer developing this way because it's easier to debug separate layers.